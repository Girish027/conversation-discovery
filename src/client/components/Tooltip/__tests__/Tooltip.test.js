import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import GenericTooltip from '../Tooltip';

describe('<GenericTooltip />', function () {
  let wrapper;

  const props = {
    content: 'The run is Queued',
    type: 'info',
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(
    <GenericTooltip
      {...propsObj}
    >
      QUEUED
    </GenericTooltip>);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for the given props', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
