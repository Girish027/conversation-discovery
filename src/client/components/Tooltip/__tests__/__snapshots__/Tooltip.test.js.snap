// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericTooltip /> Snapshots renders correctly for the given props 1`] = `
<withTheme(Tooltip)
  content="The run is Queued"
  direction="bottom"
  type="info"
>
  <div
    style={
      Object {
        "width": "max-content",
      }
    }
  >
    QUEUED
  </div>
</withTheme(Tooltip)>
`;
